docs/content/preview/explore/ysql-language-features/advanced-features/inheritance.md

```diff
@@ -0,0 +1,100 @@
+    "investment_accounts_investment_type_check" CHECK (investment_type = ANY (ARRAY['stocks'::text, 'bonds'::text, 'funds'::text]))
+Inherits: accounts
+```
+
+This schema allows for certain queries to be performed over all accounts while still preserving features unique to each account type. For example,
+
+```
+SELECT SUM(balance) FROM accounts WHERE account_id = 10;
+```
+
+Any columns added to/dropped from the parent `accounts` table are propagated to child tables so that such queries on the parent accounts table are always well formed. 
+
+However, there are certain caveats to keep in mind here.
+1. The parent table `accounts` may have its own rows that are not part of any child tables.
+2. The primary key for account_id on the parent `accounts` table does not propagate to children and has to be redefined for each child table. This is also true for foreign key constraints and non-primary key unique constaints. Special care has to be taken to maintain such constraints across parent-child hierarchies.
+
+Table inheritance can lead to complex hierarchies similar to class inheritance in object-oriented programming because a specific table can inherit from multiple parent tables and can itself be a parent table for other child tables.
+
+### Queries and updates on data
+
+SELECT and UPDATE queries on the parent table operate on a union of the parent and all child tables in the hierarchy. To restrict queries to just the specific table, the `ONLY` keyword can be used.
+
+```
+SELECT SUM(balance) FROM ONLY accounts WHERE account_id = 10;
+
+UPDATE ONLY accounts SET balance = balance + 100 WHERE account_id = 1;
+```
+
+## Schema changes
+
+1. Adding columns to or dropping columns from  or altering columns on the parent table propagates to all children in the hierarchy. The behavior of such operations can be more complex when a child table has a column of the same name and type before establishing inheritance or when a child table inherits from multiple parent tables with slightly differing definitions of a column (for example, NULL vs NOT NULL constraints on the column). For more details on the allowed differences in column definitions and the resolution of such differences in inheritance, [consult the PostgreSQL documentation](https://www.postgresql.org/docs/current/ddl-inherit.html).
+2. Adding or dropping certain kinds of constraints propagates to all children in the hierarchy. The exceptions are primary key constrains, unique constraints and foreign key constraints. For more details, consult the [PostgreSQL documentation](https://www.postgresql.org/docs/current/ddl-inherit.html)
+3. For certain schema changes, the `ONLY` keyword can be used to restrict the update to just the parent table. For example, `ALTER TABLE ONLY accounts DROP COLUMN gains` drops the column from the parent table alone while leaving it on the child tables.
+
+
+## Limitations
+
+
+https://github.com/yugabyte/yugabyte-db/issues/26094
```

Comment:
How do I make these show up with the nice syntax for github issues in our docs?

Reply:
the shortcode is in this format {{<issue 26218>}}